src/components/Search/SelectedRevisions.tsx

```diff
@@ -29,23 +46,30 @@ function SelectedRevisions({
     return selectedRep;
   });
 
+  useEffect(() => {
+    if (selectedRevisions?.length) {
+      dispatch(clearCheckedRevisions());
+    }
+  }, [dispatch]);
```

Comment:
Also it's not clear to me what this is doing. Is that to replace the call to `clearCheckedRevisions` when changing the selected revisions? I would have expected a `extraReducer` in SearchSlice to handle the `setSelectedRevisions` action (or `addSelectedRevisions` if you follow my suggeston above). See https://redux-toolkit.js.org/api/createSlice#extrareducers for more information about these, but I see there are already such reducers there, so it should be straightforward to just add another case.

Reply:
I think it would be clearer to say that `SelectedRevisions` only shows up on the Results View page and `Checked Revisions` on the Search Page. I cleared the checked revisions here so that they would not render on the Results View. The UI is different for both. 

My update: I will pass the `view` to the `SelectedRevisions` Component, eliminating the need for the whole `useEffect` above, and render `checked revisions` when the view is `'search'` and `selected revisions` when the view is `'compare-results'`. 

I will have the `SearchSlice `handle the `setSelectedRevisions` during the update process in the ResultsView.